web/packages/teleterm/src/agentCleanupDaemon/agentCleanupDaemon.test.ts

```diff
@@ -0,0 +1,171 @@
+/**
+ * Copyright 2023 Gravitational, Inc
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import childProcess from 'node:child_process';
+import path from 'node:path';
+import process from 'node:process';
+
+// inheritStdio makes it so that processes spawned during the test will inherit stdio of the process
+// running the tests. Useful for debugging.
+const inheritStdio = false;
+
+describe('agentCleanupDaemon', () => {
+  test.each([
+    {
+      name: 'terminates the agent if the parent gets terminated',
+      parentArgs: [],
+    },
+    {
+      name: 'terminates the agent if the parent gets terminated before the cleanup daemon is fully set up',
+      parentArgs: ['sendPidsImmediately'],
+    },
+    {
+      name: 'follows up SIGTERM with SIGKILL in case SIGTERM did not cause the agent to terminate',
+      parentArgs: ['sendPidsWhenReady', 'ignoreSigterm'],
+    },
+  ])('$name', async ({ parentArgs }) => {
+    await cleanupPids(async cleanupPid => {
+      const parent = childProcess.fork(
+        path.join(__dirname, 'parentTestProcess.mjs'),
+        parentArgs,
+        { stdio: (inheritStdio && 'inherit') || 'pipe' }
+      );
+      cleanupPid(parent.pid);
```

Comment:
Ah, this name confuses me, it sounds like you are cleaning pid right after creating it. Maybe `addPidToCleanup`?

Reply:
Yeah, I noticed that it might not be the best, but I didn't have a better idea. `addPidToCleanup` sounds good.